docs/developers/omnichain/bitcoin.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Bitcoin: deposit & call contracts"
+sidebar_position: 7
+---
+
+# Deposit and call zEVM contracts from Bitcoin
+
+:::tip
+
+This section is important for wallet and app developers.
+
+:::
+
+To deposit BTC into zEVM ZRC-20 contract (and optionally call a smart contract),
+the Bitcoin transaction must conform to this specifications:
+
+1. The Bitcoin transaction must have at least 2 outputs;
+2. The first output must be addressed to the TSS Bitcoin address;
+3. The second output must be a memo output, i.e. `OP_RETURN PUSH_x [DATA]`. This
+   output must be less than 80 bytes;
+4. The memo `[DATA]` is an array of bytes that encodes the recipient address of
+   this deposit into ZRC-20 or the smart contract on zEVM that will be invoked
+   by this transaction.
+5. If the purpose of this Bitcoin transaction is to only deposit BTC into the
+   BTC ZRC-20 on zEVM, then the `[DATA]` should be exactly 20 bytes long,
+   consists of an Ethereum-style address.
+6. If the purpose of this Bitcoin transaction is to deposit BTC and also use the
+   deposited amount to call a smart contract on zEVM, then the `[DATA]` field
+   must consists of a smart contract address, and a binary message that will be
+   forwarded to the said smart contract:
+   `[DATA] = [zEVM contract address (20B)] + [arbitrary binary message]`
+
+## Example 1: Deposit BTC into an account in zEVM
+
+[Here's](https://blockstream.info/testnet/tx/952d60fd9efc1aad4b87a8a7a6d57a972d49e084de8b5dc524e163216c11c04f?expand)
+an example Bitcoin transaction on Bitcoin Testnet that deposits 0.0005 tBTC
+(50000 sats) into the address (0x)6da30bfa65e85a16b05bce3846339ed2bc746316.
+
+Note the three outputs:
+
+1. sending the intended amount (50000sats) to the current TSS Bitcoin address
+   tb1qy9pqmk2pd9sv63g27jt8r657wy0d9ueeh0nqur;
+2. the memo output, encoding the recipient address on zEVM
+   (0x)6a146da30bfa65e85a16b05bce3846339ed2bc746316;
+3. change sent back to the user.
+
+If you're using ZetaChain's Hardhat
+[smart contract template](https://github.com/zeta-chain/template), you can use
+the `send-btc` task to transfer BTC:
+
+```
+npx hardhat send-btc --recipient tb1qy9pqmk2pd9sv63g27jt8r657wy0d9ueeh0nqur --amount 0.0005 --memo 6a146da30bfa65e85a16b05bce3846339ed2bc746316
+```
+
+Where `recipient` is the TSS Bitcoin address, `amount` is the amount of tBTC to
+transfer, and `memo` is the recipient address on zEVM.
+
+## Example 2: Deposit BTC and call a smart contract in zEVM
+
+:::tip
+
+In order to test with Bitcoin, you will need to use a wallet that allows setting
```

Comment:
may also want to clarify that the wallet must support a binary memo input. not essential but may help people troubleshoot.

Reply:
6cd83db